### PR TITLE
Create release process

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Created release process with the following assumptions:
   - tag naming eg. `v1.0.0`
   - release branch naming eg. `release-1.0`
   - release images naming eg. `...helm-broker:v1.0.0`
   - used changelog -> (https://github.com/github-changelog-generator/github-changelog-generator), added configuration file for it: `.github_changelog_generator`.
   - release process consists of every possible step: ```build integration-test build-image push-image charts-test release```, on `charts-test` we test a freshly builded release images.
   - create the latest release after each merge to the master branch.
   - create a makefile target for cutting the release eg. `make VERSION=v1.0.0 cut-release`.
   - create docs guide for releasing
   - release job takes ~15min to complete, BUT it may end up with an error on non-deterministic integration test or chart-test
   - [Check](https://github.com/kyma-project/helm-broker/releases) current draft releases and the release branch which were done using this process.
   - [Check](https://status.build.kyma-project.io/?job=post-release-helm-broker) the release builds logs on Prow.

**Related issue(s)**

#30

### CODE REVIEW

File: .github_changelog_generator [hunk 1]
```diff
@@ -0,0 +1,56 @@
+bugs-label=:bug: **Fixed bugs:**
```
Comment (jasiu001) [inline .github_changelog_generator]: Why do we need this file?

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,12 @@
+ROOT_PATH := $(shell pwd)
```
Comment (jasiu001) [inline Makefile]: where does the `ROOT_PATH` variable is used?

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    ```
+    git push {remote} v0.6.0
+    ```
+
+It triggers the Prow CI release job for Helm Broker.
+
+    >**NOTE:** If you want to push the tag to the upstream, run the following command:
+    >```
+    >git push upstream v0.6.0
+    >```
```
Comment (jasiu001) [inline docs/releasing.md]: IMO it better looks in one line:
` > **NOTE:** If you want to push the tag to the upstream, run the following command: ```git push upstream v0.6.0``` `

File: scripts/create_latest_tag.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
Comment (jasiu001) [inline scripts/create_latest_tag.sh]: IMO all scripts for release should be inside the hack directory (maybe `hack/release`).
If not proper note should be added to [readme](https://github.com/kyma-project/helm-broker/blob/master/README.md#project-structure)

File: scripts/get_yq.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+if [[ -z $(which yq) ]]; then
+    YAML_EDITOR=$(which yq)
+fi
+
+if [[ -z ${YAML_EDITOR} ]]; then
+    go get gopkg.in/mikefarah/yq.v2
+    YAML_EDITOR=$(which yq.v2)
+fi
+
+echo "${YAML_EDITOR}"
```
Comment (jasiu001) [inline scripts/get_yq.sh]: When I run this script on my host, the output result is `yq.v2 not found`
I think is not expected value, probably you want to install `go get gopkg.in/mikefarah/yq.v2`

File: scripts/create_latest_tag.sh (hunk 4 above)
Comment (polskikiel) [reply scripts/create_latest_tag.sh]: moved to hack/release 👍

File: hack/README.md [hunk 6]
```diff
@@ -5,3 +5,5 @@ This package contains a various scripts that are used by Helm Broker developers.
 ## Purpose
 
 This directory contains tools, such as Go fmt, Go lint, and Go vet, that help to maintain the source code compliant to Go best coding practices. It also includes utility scripts that generate code, and scripts executed on CI pipelines.
+
+The `release` directory contains the scripts which are essential for the release process.
```
Comment (klaudiagrz) [inline hack/README.md]: I'd add another section called `## Project structure` with the directory tree, such as here:
https://github.com/kyma-project/kyma#project-structure

Please also fix the first sentence of this document:
This package contains ~~a~~ various scripts that are used by Helm Broker developers.

File: README.md [hunk 7]
```diff
@@ -164,6 +164,20 @@ Run integration tests using this command:
 make integration-test
 ```
 
+### Update the chart's images tag
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
### Update chart's images tag
```

File: README.md [hunk 8]
```diff
@@ -164,6 +164,20 @@ Run integration tests using this command:
...
+
+Change the chart's tags version using the command:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
To change the chart's tags version, run this command:
```

File: README.md [hunk 9]
```diff
@@ -164,6 +164,20 @@ Run integration tests using this command:
...
+
+```bash
+make VERSION=v0.0.1 DIR=/pr tag-chart-images
+```
+
+Above command overrides the images tag in the `charts/helm-broker/values.yaml` file to:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
This command overrides the images tag in the `charts/helm-broker/values.yaml` file to:
```

File: docs/releasing.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+Every change in the master branch which passes all checks creates the `latest` tag and triggers the release.
+
+Every tag pushed to the repository triggers a new release.
+
+### Create a release from the master branch
+
+Follow these steps to create a new `helm-broker` release `v0.6.0`:
+
+1. Checkout on master and pull the latest changes:
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
1. Checkout the `master` branch and pull the latest changes:
```

File: docs/releasing.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    ```
+    git checkout master
+    git pull
+    ```
+2. Prepare the release:
+
+    ```
+    make VERSION=v0.6.0 cut-release
+    ```
+Above command do the following things:
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
This command does the following:
```

File: docs/releasing.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  - changes the images tag in the `charts/helm-broker/values.yaml` to `v0.6.0`
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
  - Changes the images tag in the `charts/helm-broker/values.yaml` to `v0.6.0`
```

File: docs/releasing.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+  - creates a commit with the changes
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
  - Creates a commit with the changes
```

File: docs/releasing.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+  - creates a `v0.6.0` tag
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
  - Creates the `v0.6.0` tag
```

File: docs/releasing.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+3. Push the release:
+
+    ```
+    git push {remote} v0.6.0
+    ```
+
+It triggers the Prow CI release job for Helm Broker.
+
+> **NOTE:** If you want to push the tag to the upstream, run the following command: ```git push upstream v0.6.0```
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
> **NOTE:** If you want to push the tag to the upstream, run `git push upstream v0.6.0`.
```

File: docs/releasing.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+4. After Prow CI job finish, go to the **releases** tab where the new release appears. The new branch is available in the **branches** tab.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
4. After Prow CI job is finished, go to the **Releases** tab where the new release appears. The new branch is available in the **Branches** tab.
```

File: docs/releasing.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+1. Checkout the release branch with the latest changes. For example:
+
+    ```
+    git fetch {remote}
+    git checkout release-0.6
+    git pull
+    ```
+
+2. Commit your changes to the release branch `release-0.6`.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
2. Commit your changes to the `release-0.6` release branch.
```

File: docs/releasing.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+>**NOTE:** In order to have a complete changelog, create a Pull Requests to the release branch.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
>**NOTE:** In order to have a complete changelog, create a Pull Request to the release branch.
```

File: docs/releasing.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+3. Prepare the release:
+
+    ```
+    make VERSION=v0.6.1 cut-release
+    ```
+    Above command do the following things:
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
    This command does the following:
```

File: docs/releasing.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+      - changes the images tag in the `charts/helm-broker/values.yaml` to `v0.6.1`
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
      - Changes the images tag in the `charts/helm-broker/values.yaml` to `v0.6.1`
```

File: docs/releasing.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+      - creates a commit with the changes
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
      - Creates a commit with the changes
```

File: docs/releasing.md [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+      - creates a `v0.6.1` tag
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
      - Creates the `v0.6.1` tag 
```

File: docs/releasing.md [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+      - creates a commit with the changes
+      - creates a `v0.6.1` tag 
+
+4. Push your changes with the tag:
+
+    ```
+    git push {remote} v0.6.1
+    ```
+
+5. After Prow CI job finish, go to the **releases** tab where the new `v0.6.1` release appears.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
5. After Prow CI job is finished, go to the **Releases** tab where the new `v0.6.1` release appears.
```

File: docs/releasing.md [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Changelog functionality
+
+Every time you create a new release, the changelog file is generated and it's placed in the release body.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
Every time you create a new release, the changelog file is generated. It is automatically added to the release body.
```

File: docs/releasing.md [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+
+Every changelog file contains a list of pull requests and issues from the period between the releases. It's generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator).
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
Every changelog file contains a list of Pull Requests and issues from the period between the releases. It is generated by the [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator).
```

File: docs/releasing.md [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+You can configure the changelog generator in the `.github_changelog_generator` file according to the [github_changelog_generator](https://github.com/github-changelog-generator/github-changelog-generator#params-file) documentation.
```
Comment (klaudiagrz) [inline docs/releasing.md]:
```suggestion
If you want to configure the changelog generator in the `.github_changelog_generator` file, read [this](https://github.com/github-changelog-generator/github-changelog-generator#params-file) document.
```

**Pull request merged**